File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        sort(intervals.begin(),intervals.end());
       // int curr = intervals[0];
        vector<int> temp = intervals[0];
        for(auto it : intervals){
            if(it[0] <= temp[1]){
                temp[1] = max(it[1],temp[1]);
            }else{
                ans.push_back(temp);
                temp = it;
            }
        }
        ans.push_back(temp);
        return ans;
    }  
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        LinkedList<int[]> merge = new LinkedList<>();
        for(int[] it : intervals){
            if(merge.isEmpty() ||merge.getLast()[1]<it[0]){
                merge.add(it);
            } 
            else{
                merge.getLast()[1] = Math.max(merge.getLast()[1],it[1]);
            }
        }
         
        return merge.toArray(new int[merge.size()][]);
        
        
    }
}